tap_shiphero/sync.py

```diff
@@ -0,0 +1,255 @@
+
+            if num_results > 0:
+                if num_results == limit:
+                    page += 1
+
+                persist_records(catalog, stream_id, records)
+        
+        set_bookmark(state, stream_id, updated_from.isoformat())
+        updated_from = updated_to
+
+def sync_vendors(client, catalog):
+    stream_id = 'vendors'
+
+    LOGGER.info('Sycing all vendors')
+
+    write_schema(catalog, stream_id)
+
+    data = client.get(
+        '/list-vendors/',
+        endpoint=stream_id)
+
+    records = data['vendors']
+
+    persist_records(catalog, stream_id, records)
+
+def sync_shipments(client, catalog, state, start_date, end_date):
+    stream_id = 'shipments'
+
+    write_schema(catalog, stream_id)
+
+    last_date = get_bookmark(state, stream_id, start_date)
+
+    if end_date:
+        end_date = pendulum.parse(end_date)
+    else:
+        end_date = pendulum.now('UTC')
+
+    updated_from = pendulum.parse(last_date)
+
+    if updated_from > end_date:
+        raise Exception('Shipments start_date is greater than end_date')
+
+    limit = 100
+    while updated_from < end_date:
+        page = 1
+        num_results = None
+        updated_to = updated_from.add(days=1)
+        while is_next_page(limit, num_results):
+            LOGGER.info('Sycing shipments from: {} to: {} - page {}'.format(
+                updated_from.to_date_string(),
+                updated_to.to_date_string(),
+                page))
+            data = client.get(
+                '/get-shipments/',
+                params={
+                    'from': updated_from.to_date_string(),
+                    'to': updated_to.to_date_string(),
+                    'page': page,
+                    'filter_on': 'shipment',
+                    'all_orders': 1
+                },
+                endpoint=stream_id)
+
+            raw_shipments = data.get('orders', {}).get('results', {})
+            records = []
+            for shipment_id, shipment in raw_shipments.items():
+                records.append({'shipment_id': shipment_id, **shipment})
+
+            num_results = len(records)
+
+            if num_results > 0:
+                if num_results == limit:
+                    page += 1
+
+                persist_records(catalog, stream_id, records)
+        
+        set_bookmark(state, stream_id, updated_from.isoformat())
+        updated_from = updated_to
+
+def should_sync_stream(last_stream, selected_streams, stream_name):
+    if last_stream == stream_name or \
+       (last_stream is None and stream_name in selected_streams):
+        return True
+    return False
+
+def set_current_stream(state, stream_name):
+    state['current_stream'] = stream_name
+    singer.write_state(state)
+
+def sync(client, catalog, state, start_date, end_date):
+    selected_streams = get_selected_streams(catalog)
+
+    last_stream = state.get('current_stream')
+
+    if should_sync_stream(last_stream, selected_streams, 'products'):
```

Comment:
We think there's a bug in this implementation: If a stream is resumed it appears that it will be the only stream that syncs during the course of that run.

The ideal implementation should be a `for` loop here where the `current_stream` gets shuffled to the first entry in the array so it resumes first. See https://github.com/singer-io/tap-shopify/blob/master/tap_shopify/__init__.py#L120-L124